### DISCUSSION TITLE
Using certain characters in the code `fence` node

### DISCUSSION THREAD

**Discussion by jayanratna** [markdoc/markdoc#235] [Q&A] [answered]
I am writing documentation using Next.js, that involves adding code examples for a templating language called [Nunjucks](https://mozilla.github.io/nunjucks). To add these examples I would create a new code `fence` and set the language to `nunjucks` and obviously add the code example within... but doing this causes validation errors to occur. Right off the bat Nunjucks has syntax that is very similar to the Markdoc syntax and I can see why there would be confusion by the validation system. I thought I would checkout the Markdoc `docs` repo to see how its done there but then I realised that Markdoc wouldn't fail validation on its own syntax, which is probably why it works as intended there.

**Comment by mfix-stripe** [comment, marked as answer]
@jayanratna can you share an example snippet that is failing for you? 

I imagine you can work around this by setting `{% process=false %}` on your code fence:

````md
```md {% process=false %}
{% extends "base.html" %}

{% block header %}
<h1>{{ title }}</h1>
{% endblock %}

{% block content %}
<ul>
  {% for name, item in items %}
  <li>{{ name }}: {{ item }}</li>
  {% endfor %}
</ul>
{% endblock %}
```
````

Here is [what the AST looks like](https://markdoc.dev/sandbox?c=YGBgbWQgeyUgcHJvY2Vzcz1mYWxzZSAlfQp7JSBleHRlbmRzICJiYXNlLmh0bWwiICV9Cgp7JSBibG9jayBoZWFkZXIgJX0KPGgxPnt7IHRpdGxlIH19PC9oMT4KeyUgZW5kYmxvY2sgJX0KCnslIGJsb2NrIGNvbnRlbnQgJX0KPHVsPgogIHslIGZvciBuYW1lLCBpdGVtIGluIGl0ZW1zICV9CiAgPGxpPnt7IG5hbWUgfX06IHt7IGl0ZW0gfX08L2xpPgogIHslIGVuZGZvciAlfQo8L3VsPgp7JSBlbmRibG9jayAlfQpgYGA%253D&mode=ast):

![image](https://user-images.githubusercontent.com/62121649/193316187-fe4b18e0-8bb4-4a52-8786-7b879d516a8e.png)

**Reply by jayan-blutui** [reply]
Thank you! That worked like a charm. Had to update my custom `fence` node to get it to work like the default, as I'm using a React component instance for `render`
